isis/tests/FunctionalTestsFindGaps.cpp

```diff
@@ -0,0 +1,195 @@
+#include <QTemporaryDir>
+
+#include "findgaps.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Histogram.h"
+#include "LineManager.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/findgaps.xml").expanded();
+
+// Tests a basic gap detection
+TEST_F( SmallCube, FindGapsDefault )
+{
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/findgaps_out.cub";
+  QString logFileName = prefix.path() + "/findgaps_log.txt";
+  QVector<QString> args = {"from=" + testCube->fileName(),
+                          "to=" + cubeFileName,
+                           "log=" + logFileName,
+                           "above=1", "below=1" };
+  UserInterface options(APP_XML, args);
+
+  // fill with nulls from line 4 to line 5 through all bands
+  LineManager line(*testCube);
+  int lineNum = 0;
+  int startGap = 4;
+  int endGap = 5;
+  for(line.begin(); !line.end(); line.next()) {
+    for(int i = 0; i < line.size(); i++) {
+      if(line.Line() == startGap || line.Line() == endGap)
+      {
+        line[i] = NULL8;
+      }
+    }
+    lineNum++;
+    if(line.Line() == startGap || line.Line() == endGap)
+    {
+      testCube->write(line);
+    }
```

Comment:
Put the if before the loop across the buffer so this is:

```
for line in cube
  if line == start or line == end
    for pixel in line:
      fill with nulls
    write line
```

That should save a lot of ifs and loops